### ISSUE TITLE
Issue parsing colons on Linux

### ISSUE THREAD

**Issue by farski** [aws-cloudformation/cfn-lint#1984]
*cfn-lint version: 0.48.2 (`cfn-lint --version`)*

*Description of issue.*

When I lint [this file](https://github.com/PRX/Infrastructure/blob/master/stacks/shared-vpc.yml#L38) on my Mac using 0.48.2, I get no errors. When the same file is linted using 0.48.2 in CodeBuild (aws/codebuild/amazonlinux2-x86_64-standard:3.0) it returns:

```
E0000 found unexpected ':'
./stacks/shared-vpc.yml:38:21
```

The offending line is:
`- { Key: prx:meta:tagging-version, Value: "2021-04-07" }`

CloudFormation has no issues with the code (it creates those tags as expected). Quoting the values makes the linter happy, even in CodeBuild, but it shouldn't be necessary, and since the results are inconsistent between platforms it's hard/annoying to catch them during development. Also it's a false positive, so it's mainly just noise during CI.

**Comment by PatMyron** [comment]
Interesting, wasn't able to reproduce on Ubuntu either. Wonder if it's specific to Amazon Linux

**Comment by kddejong** [comment]
Just for reference we did find some oddities in Alpine systems too.  https://github.com/aws-cloudformation/cfn-python-lint/issues/666  not sure if this is related or not but its worth noting we try to use the C parsers when available for speed reasons.

**Comment by kddejong** [comment]
Possible references that we need to investigate.  We need to start by knowing if it ends up using the Cparser or not and then what version of that library it may be using. https://github.com/yaml/libyaml/pull/28 and https://github.com/yaml/pyyaml/pull/45

**Comment by farski** [comment]
Seems like the most recent release has resolved this

**Comment by PatMyron** [comment]
maybe https://github.com/aws-cloudformation/cfn-python-lint/pull/1992?
